Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP.py

```diff
@@ -205,31 +295,256 @@ def test(client):
+        outputs_prefix='DLP.exemption',
+        outputs_key_field='eligible',
+        outputs=result
+    )
+
+
+def slack_bot_message(args: dict, params: dict):
+    message_template = params.get('dlp_slack_message', '')
+    template = Template(message_template)
+    message = template.substitute(
+        user=args.get('user'),
+        file_name=args.get('file_name'),
+        data_profile_name=args.get('data_profile_name'),
+        app_name=args.get('app_name'),
+        snippets=args.get('snippets', ""))
+    result = {
+        'message': message
+    }
+    return CommandResults(
+        outputs_prefix='DLP.slack_message',
+        outputs_key_field='slack_message',
+        outputs=result
+    )
+
+
+def fetch_incidents_command() -> List[Dict]:
+    """
+    Fetch incidents implemented, for mapping purposes only.
+    Returns list of samples saved by long running execution.
+
+    Returns:
+        (List[Dict]): List of incidents samples.
+    """
+    ctx = get_integration_context()
+    return ctx.get('samples', [])
+
+
+def reset_last_run_command() -> str:
+    """
+    Puts the reset flag inside integration context.
+    Returns:
+        (str): 'fetch-incidents was reset successfully'.
+    """
+    ctx = get_integration_context()
+    ctx[RESET_KEY] = 'true'
+    set_to_integration_context_with_retries(ctx)
+    return 'fetch-incidents was reset successfully.'
+
+
 def main():
     """ Main Function"""
     try:
         demisto.info('Command is %s' % (demisto.command(),))
         params = demisto.params()
         access_token = params.get('access_token')
         refresh_token = params.get('refresh_token')
-
-        client = Client(refresh_token, access_token, params.get('insecure'), params.get('proxy'))
-
+        url = get_base_url(params)
+        client = Client(url, refresh_token, access_token, params.get('insecure'), params.get('proxy'))
+        args = demisto.args()
         if demisto.command() == 'pan-dlp-get-report':
-            args = demisto.args()
             report_id = args.get('report_id')
             fetch_snippets = argToBoolean(args.get('fetch_snippets'))
             report_json, status_code = client.get_dlp_report(report_id, fetch_snippets)
-            parse_dlp_report(report_json)
-
-        if demisto.command() == "test-module":
+            return_results(parse_dlp_report(report_json))
+        elif demisto.command() == 'fetch-incidents':
+            demisto.incidents(fetch_incidents_command())
+        elif demisto.command() == 'long-running-execution':
+            long_running_execution_command(params)
+        elif demisto.command() == 'pan-dlp-update-incident':
+            return_results(update_incident(client, args))
+        elif demisto.command() == 'pan-dlp-exemption-eligible':
+            return_results(exemption_eligible(args, params))
+        elif demisto.command() == 'pan-dlp-slack-message':
+            return_results(slack_bot_message(args, params))
```

Comment:
Please add to all of these command functions the `_command` suffix.

Reply:
Add _command suffix to all the command functions.